zyte_common_items/pipelines.py

```diff
@@ -53,3 +57,49 @@ def __init__(self):
 
     def process_item(self, item, spider):
         return ae.downgrade(item)
+
+
+class DropItem(Exception):
+    """Drop item from the item pipeline."""
+
+    pass
+
+
+class DropLowProbabilityItemPipeline:
+    """Drop item with a probability less than threshold."""
+
+    def __init__(self, crawler):
+        self.stats = crawler.stats
+        self.thresholds = {}
+        self.init_thresholds(crawler.spider)
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        o = cls(crawler)
+        return o
+
+    def init_thresholds(self, spider):
+        thresholds_settings = spider.settings.get("ITEM_PROBABILITY_THRESHOLDS", {})
+        for item, threshold in thresholds_settings.items():
+            item_type = item if isinstance(item, str) else global_object_name(item)
+            self.thresholds[item_type] = threshold
+
+    def get_threshold(self, item, spider):
+        return self.thresholds.get(
+            global_object_name(item.__class__), DEFAULT_ITEM_PROBABILITY_THRESHOLD
+        )
+
+    async def process_item(self, item, spider):
```

Comment:
```suggestion
    def process_item(self, item, spider):
```

Reply:
@Gallaecio just wondering - why do we need to use not async here?

Reply:
Is not that we need not to use async, is that we do not need to use async. There is no penalty to using either I think, my thinking here is that using `async` could be misleading, making people think it does async stuff. It also makes it harder to reuse this method from sync code. No strong opinion, though, it is OK to leave it as it is.

Reply:
Thanks, fixed already, you are right, let's leave it more clear without async.